service-userlayer/src/main/java/org/oskari/map/userlayer/service/UserLayerDataService.java

```diff
@@ -70,85 +64,88 @@ public UserLayer storeUserData(GeoJsonWorker gjsWorker, User user, Map<String, S
-            if (fparams.containsKey(KEY_DESC)) userLayer.setLayer_desc(fparams.get(KEY_DESC));
-            if (fparams.containsKey(KEY_SOURCE)) userLayer.setLayer_source(fparams.get(KEY_SOURCE));
+            if (fparams.containsKey(KEY_NAME)) {
+                userLayer.setLayer_name(fparams.get(KEY_NAME));
+            }
+            if (fparams.containsKey(KEY_DESC)) {
+                userLayer.setLayer_desc(fparams.get(KEY_DESC));
+            }
+            if (fparams.containsKey(KEY_SOURCE)) {
+                userLayer.setLayer_source(fparams.get(KEY_SOURCE));
+            }
 
             //get userLayerData list
-            userLayerDataList = this.getUserLayerData(gjsWorker.getGeoJson(), user, userLayer);
+            List<UserLayerData> userLayerDataList = getUserLayerData(fc, user, userLayer);
 
             if (userLayerDataList.isEmpty()){
                 throw new ServiceException ("no_features");
             }
             //insert layer, style and data in one transaction
             int count = userLayerService.insertUserLayer(userLayer, style, userLayerDataList);
             log.info("stored:",count, "rows from", userLayer.getFeatures_count(), "features and skipped:",userLayer.getFeatures_skipped());
+            return userLayer;
         } catch (Exception e) {
             log.error(e, "Unable to store user layer  data");
             throw new ServiceException ("unable_to_store_data");
         }
-        return userLayer;
     }
 
+    private List<UserLayerData> getUserLayerData(SimpleFeatureCollection fc, User user, UserLayer userLayer) throws ServiceException{
+        try (SimpleFeatureIterator it = fc.features()) {
+            final List<UserLayerData> userLayerDataList = new ArrayList<>();
+            final String uuid = user.getUuid();
 
-    /**
-     * @param geoJson import data in geojson format
-     * @param user    oskari user
-     * @param userLayer      user layer id in user_layer table
-     * @return
-     */
-    public List <UserLayerData> getUserLayerData(JSONObject geoJson, User user, UserLayer userLayer) throws ServiceException{
+            int count = 0;
+            int noGeometry = 0;
 
-
-        int count = 0;
-        int noGeometry = 0;
-        List <UserLayerData> userLayerDataList = new ArrayList<UserLayerData>();
-        String uuid = user.getUuid();
-
-        try {
-            final JSONArray geofeas = geoJson.getJSONArray("features");
-
-            // Loop json features and fix to user_layer_data structure
-            for (int i = 0; i < geofeas.length(); i++) {
-                JSONObject geofea = geofeas.optJSONObject(i);
-                if (geofea == null){
-                    continue;
-                }
-
-                if (!geofea.has("geometry") || geofea.optJSONObject("geometry")== null) {
+            while (it.hasNext()) {
+                SimpleFeature f = it.next();
+                if (f.getDefaultGeometry() == null) {
```

Comment:
Here's the null defaultGeometry handling that I was requesting earlier. It's fine to handle it here.